Mono.Counters/NativeCounter.cs

```diff
@@ -34,9 +34,21 @@ internal object GetValue(byte[] array)
                     case CounterType.IntPtr:
                         return MemoryMarshal.Read<IntPtr>(array);
                     case CounterType.Long:
-                        return MemoryMarshal.Read<long>(array);
+                        long value = MemoryMarshal.Read<long>(array);
+                        if (Unit.Equals(CounterUnit.Time))
+                            return TimeSpan.FromTicks(value / 10000);
+                        return value;
                     case CounterType.ULong:
-                        return MemoryMarshal.Read<ulong>(array);
+                        ulong value2 = MemoryMarshal.Read<ulong>(array);
+                        if (Unit.Equals(CounterUnit.Time))
+                        {
+                            value2 /= 10000;
+                            if (value2 < long.MaxValue)
+                                return TimeSpan.FromTicks((long)value2);
+                            else
+                                return value2;
```

Comment:
It's most likely not possible for this condition to fail. `ulong.MaxValue` is actually `long.MaxValue * 2`. So just hard casting without the check is safe.

```suggestion
                            value2 /= 10 * 1000; // Mono uses 100ns as the baseline.
                            return TimeSpan.FromTicks((long)value2);
```